tesseract/tesseract_environment/test/tesseract_environment_unit.cpp

```diff
@@ -324,172 +530,52 @@ void runCloneTest(const tesseract_environment::Environment::ConstPtr& env)
-
-  // Set Active contact manager
-  EXPECT_TRUE(env->setActiveDiscreteContactManager(tesseract_collision_bullet::BulletDiscreteBVHManager::name()));
-  EXPECT_TRUE(env->setActiveContinuousContactManager(tesseract_collision_bullet::BulletCastBVHManager::name()));
-
-  runChangeJointOriginTest(env);
+  runChangeJointOriginTest<KDLStateSolver>();
+  runChangeJointOriginTest<OFKTStateSolver>();
 }
 
 TEST(TesseractEnvironmentUnit, EnvCurrentStatePreservedWhenEnvChanges)  // NOLINT
 {
-  SceneGraph::Ptr scene_graph = getSceneGraph();
-  EXPECT_TRUE(scene_graph != nullptr);
-
-  auto env = std::make_shared<Environment>();
-  EXPECT_TRUE(env != nullptr);
-
-  bool success = env->init<KDLStateSolver>(*scene_graph);
-  EXPECT_TRUE(success);
-
-  runCurrentStatePreservedWhenEnvChangesTest(env);
+  runCurrentStatePreservedWhenEnvChangesTest<KDLStateSolver>();
+  runCurrentStatePreservedWhenEnvChangesTest<OFKTStateSolver>();
 }
 
-TEST(TesseractEnvironmentUnit, addSceneGraph)
+TEST(TesseractEnvironmentUnit, EnvAddSceneGraph)
 {
-  SceneGraph::Ptr scene_graph = getSceneGraph();
-  EXPECT_TRUE(scene_graph != nullptr);
-
-  SceneGraph::Ptr subgraph = std::make_shared<SceneGraph>();
-  subgraph->setName("subgraph");
-
-  Environment env;
-
-  bool success = env.init<KDLStateSolver>(*scene_graph);
-  EXPECT_TRUE(success);
-
-  // Adding an empty scene graph which should fail
-  EXPECT_FALSE(env.addSceneGraph(*subgraph));
-
-  // Now add a link to empty environment
-  Link link("subgraph_base_link");
-  subgraph->addLink(std::move(link));
-
-  EXPECT_TRUE(env.addSceneGraph(*subgraph));
-  EXPECT_TRUE(env.getJoint("subgraph_joint") != nullptr);
-  EXPECT_TRUE(env.getLink("subgraph_base_link") != nullptr);
-
-  // Adding twice with the same name should fail
-  EXPECT_FALSE(env.addSceneGraph(*subgraph));
-  EXPECT_TRUE(env.addSceneGraph(*subgraph, "prefix_"));
-  EXPECT_TRUE(env.getJoint("prefix_subgraph_joint") != nullptr);
-  EXPECT_TRUE(env.getLink("prefix_subgraph_base_link") != nullptr);
+  runAddSceneGraphTest<KDLStateSolver>();
+  runAddSceneGraphTest<OFKTStateSolver>();
 }
 
-TEST(TesseractEnvironmentUnit, applyCommands)  // NOLINT
+TEST(TesseractEnvironmentUnit, EnvApplyCommands)  // NOLINT
 {
-  SceneGraph::Ptr scene_graph = getSceneGraph();
-  EXPECT_TRUE(scene_graph != nullptr);
-
-  auto env = std::make_shared<Environment>();
-  EXPECT_TRUE(env != nullptr);
-
-  bool success = env->init<KDLStateSolver>(*scene_graph);
-  EXPECT_TRUE(success);
-
-  runApplyCommandsTest(env);
+  runApplyCommandsTest<KDLStateSolver>();
+  runApplyCommandsTest<OFKTStateSolver>();
 }
 
 TEST(TesseractEnvironmentUnit, clone)  // NOLINT
 {
-  SceneGraph::Ptr scene_graph = getSceneGraph();
-  EXPECT_TRUE(scene_graph != nullptr);
-
-  auto env = std::make_shared<Environment>();
-  EXPECT_TRUE(env != nullptr);
-
-  bool success = env->init<KDLStateSolver>(*scene_graph);
-  EXPECT_TRUE(success);
-
-  runCloneTest(env);
+  runCloneTest<KDLStateSolver>();
```

Comment:
Should this be changed to EnvClone to match?

Reply:
done.